commands/util/charts.js

```diff
@@ -1,28 +1,151 @@
 import { SlashCommandBuilder, EmbedBuilder } from 'discord.js';
+import axios from 'axios';
+import cheerio from 'cheerio';
+
+
+// const fetchCurrentCycle = async () => {
+//   try {
+//     const response = await axios.get(`${baseUrl}`);
+//     const $ = cheerio.load(response.data);
+//     console.log('Successfully fetched the current cycle.');
+//     return $('select#cycle > option:contains(Current)').val();
+//   } catch (error) {
+//     console.error('Error fetching the current cycle:', error);
+//     throw error;
+//   }
+// };
+
+const listOne = async (icao, chartType) => {
+  try {
+    const searchCycle = '2403'
+    const url = `https://www.faa.gov/air_traffic/flight_info/aeronav/digital_products/dtpp/search/results/?cycle=${searchCycle}&ident=${icao}&sort=type&dir=asc`;
+    const response = await axios.get(url);
+    console.log(`Successfully fetched diagrams for ${icao}.`);
+    return parse(response.data, chartType);
+  } catch (error) {
+    console.error(`Error fetching diagrams for ${icao}:`, error);
+    throw error;
+  }
+};
+
+const parse = (html, chartType) => {
+  const $ = cheerio.load(html);
+  const $resultsTable = $('#resultsTable');
+
+  if (!$resultsTable.html()) {
+    console.error('Unable to parse the #resultsTable page element');
+    return [];
+  }
+
+  const results = $resultsTable
+    .find('tr')
+    .toArray()
+    .filter(row => isDiagramRow($(row), chartType))
+    .map(row => extractRow($(row)))
+    .filter(x => !!x);
+
+  console.log('Successfully parsed diagram data.');
+  return results;
+};
+
+const text = ($row, columnIndex) =>
+  $row
+    .find(`td:nth-child(${columnIndex})`)
+    .text()
+    .trim();
+
+const link = ($row, columnIndex) =>
+  $row
+    .find(`td:nth-child(${columnIndex})`)
+    .find('a')
+    .attr('href');
+
+const extractRow = ($row) => {
+  return {
+    state: text($row, 1),
+    city: text($row, 2),
+    airport: text($row, 3),
+    ident: text($row, 4),
+    vol: text($row, 5),
+    flag: text($row, 6),
+    procedure: {
+      name: text($row, 8),
+      url: link($row, 8)
+    },
+    compare: {
+      name: text($row, 9),
+      url: link($row, 9)
+    }
+  };
+};
+
+const isDiagramRow = ($row, chartType) => {
+  if (!chartType) return true;
+  return text($row, 7) === chartType;
+};
+
+const airportDiagrams = async (icaos, chartType, options = {}) => {
```

Comment:
instead of having this be a const (which turns into a function after the const is declared, why not just make it a function itself. The usage of it is the same as what you have now, and it looks cleaner.